url27a7ae3508_serkan_ozal_ocean_of_memories_tgz-pJ8-com_zeroturnaround_rebellabs_oceanofmemories_article1_objectlayout_ObjectMemoryLayoutDemoJ8/src/com/zeroturnaround/rebellabs/oceanofmemories/common/util/ClasspathUtil.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * @author SERKAN OZAL
+ *         
+ *         E-Mail: <a href="mailto:serkanozal86@hotmail.com">serkanozal86@hotmail.com</a>
+ *         GitHub: <a>https://github.com/serkan-ozal</a>
+ */
+
+package com.zeroturnaround.rebellabs.oceanofmemories.common.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.net.URLConnection;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+
+public class ClasspathUtil {
+
+    private static final Logger logger = Logger.getLogger(ClasspathUtil.class);
+
+    private static Set<URL> classpathUrls;
+    private static String fullClasspath;
+
+    static {
+        init();
+    }
+
+    private ClasspathUtil() {
+
+    }
+
+    public static Set<URL> getClasspathUrls() {
+        return classpathUrls;
+    }
+
+    public static String getFullClasspath() {
+        return fullClasspath;
+    }
+
+    private static void init() {
+        classpathUrls = findClasspathUrls();
+        logger.info("Found classpath URL list: " + classpathUrls);
+        StringBuilder classpathBuilder = new StringBuilder();
+        if (classpathUrls != null) {
+            for (URL url : classpathUrls) {
+                classpathBuilder.append(url.getPath()).append(
+                        File.pathSeparator);
+            }
+        }
+        fullClasspath = classpathBuilder.toString().replace("%20", " ");
+    }
+
+    private static Set<URL> findClasspathUrls() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 26. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 64-91, lines 94-99, lines 103-108, lines 110-115 into separate methods.